src/NumericalAlgorithms/FiniteDifference/Reconstruct.tpp

```diff
@@ -211,21 +283,40 @@ void reconstruct_impl(const gsl::not_null<gsl::span<double>*> recons_upper,
     const auto upper_lower_and_order = Reconstructor::pointwise(
         q.data() + ghost_zone_for_stencil, 1, args_for_reconstructor...);
     (*recons_upper)[recons_slice_offset + volume_extents[0]] =
-        upper_lower_and_order[0];
+        get<0>(upper_lower_and_order);
+    if constexpr (ReturnReconstructionOrder and
+                  std::tuple_size<
+                      std::decay_t<decltype(upper_lower_and_order)>>::value >
+                      2) {
+      // recons_order_slice_offset + 1 (for left neighbor) + (volume_extents-1)
```

Comment:
is there a way to factor out this code block that is repeated five times?  Also I don't understand the comments in the last two